include/status-code/errored_status_code.hpp

```diff
@@ -204,34 +204,38 @@ template <class ErasedType> class errored_status_code<erased<ErasedType>> : publ
 
   /***** KEEP THESE IN SYNC WITH STATUS_CODE *****/
   //! Implicit copy construction from any other status code if its value type is trivially copyable and it would fit into our storage
-  SYSTEM_ERROR2_TEMPLATE(class DomainType)                                                                        //
-  SYSTEM_ERROR2_TREQUIRES(SYSTEM_ERROR2_TPRED(std::is_trivially_copyable<typename DomainType::value_type>::value  //
-                                              &&detail::type_erasure_is_safe<value_type, typename DomainType::value_type>::value))
+  SYSTEM_ERROR2_TEMPLATE(class DomainType)                                                                                        //
+  SYSTEM_ERROR2_TREQUIRES(SYSTEM_ERROR2_TPRED(std::is_trivially_copyable<typename DomainType::value_type>::value                  //
+                                              &&detail::type_erasure_is_safe<value_type, typename DomainType::value_type>::value  //
+                                              && (sizeof(status_code<DomainType>) - sizeof(detail::status_code_storage<DomainType>) == 0)))
```

Comment:
I always feel uncomfortable when I see subtraction of unsigned values - I get queasy feelings of potential future surprise unsigned overflow. Any reason you chose this over simple `==`?

Reply:
No real reason, it's just an artifact of my thinking process. Will fix.